### PR TITLE
Add CDP Documentation

### PR DESCRIPTION
**Pull request by josipbudzaki-sl**
### Description
Adding CDP documentation

### Motivation and Context
document CDP usage

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/web-apps/automated-testing/cdp.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+values={[
+{label: 'Java', value: 'Java'},
+{label: 'Python', value: 'Python'},
+{label: 'WebdriverIO', value: 'WebdriverIO'},
+]}>
+
+<TabItem value="Java">
+
+```java
+// Add selenium example
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp.md:45]: example is missing

File: docs/web-apps/automated-testing/cdp.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+SELENIUM_REMOTE_URL="https://{SAUCE_USERNAME}:{SAUCE_ACCESS_KEY}@ondemand.eu-central-1.saucelabs.com:443/wd/hub"
+SELENIUM_REMOTE_CAPABILITIES='{"browserName": "chrome", "sauce:options": {"devTools": "true", "name": "My Playwright Selenium Grid CDP Test"}}'
+```
+
+</TabItem>
+</Tabs>
+
+For more information, please refer to the [Playwright Selenium Grid documentation](https://playwright.dev/docs/selenium-grid)
+
+### 3. Using Playwright (WebSocket)
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp.md:113]: this options is fairly limited, not even supporting latest browsers, i would omit it for now.

File: docs/web-apps/automated-testing/cdp.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
+---
+id: cdp-bidi
+title: CDP / BiDi on Sauce Labs
+sidebar_label: Using CDP / BiDi
+description: An introduction to CDP / BiDi usage within Sauce Labs
+---
+
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp.md:7]: BETA Warning missing

File: docs/web-apps/automated-testing/cdp-bidi/examples.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+id: examples
+title: Examples
+sidebar_label: Examples
+description: Examples for both CDP API and BiDi API usage
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp-bidi/examples.md:11]: can we add here a statement that this is in BETA

File: docs/web-apps/automated-testing/cdp-bidi/examples.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    auth = {'authorization': 'Basic ' + credentials}
+
+    await connection.session.execute(connection.devtools.network.set_extra_http_headers(Headers(auth)))
+```
+
+</TabItem>
+<TabItem value="WebdriverIO">
+
+```javascript
+// Add Example
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp-bidi/examples.md:128]: example missing

File: docs/web-apps/automated-testing/cdp-bidi/examples.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+        log = Log(driver, session)
+
+        async with log.add_listener(Console.ALL) as messages:
+```
+
+</TabItem>
+<TabItem value="WebdriverIO">
+
+```javascript
+// Add Example
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp-bidi/examples.md:174]: another example missing

File: docs/web-apps/automated-testing/cdp-bidi/examples.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+            content_types.append(event.response.headers["content-type"])
+            if "text/html; charset=utf-8" in content_types:
+                break
+```
+
+</TabItem>
+<TabItem value="WebdriverIO">
+
+```javascript
+// Add Example
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp-bidi/examples.md:232]: ...

File: docs/web-apps/automated-testing/cdp.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+values={[
+{label: 'Java', value: 'Java'},
+{label: 'Python', value: 'Python'},
+{label: 'WebdriverIO', value: 'WebdriverIO'},
+]}>
+
+<TabItem value="Java">
+
+```java
+// Add Playwright example
```
Comment (aluedeke) [inline docs/web-apps/automated-testing/cdp.md:128]: ...

**Review by kristofmuhi — APPROVED** [review]

**Pull request merged**